### PR TITLE
Fix shopify order adjustment data mapping issue

### PR DESCRIPTION
**Pull request by RolandHewage**
# Description

Fix shopify order adjustment data mapping issue

Fixes # https://github.com/wso2-enterprise/internal-support-ballerina/issues/417

One line release note: 
- Fix shopify order adjustment data mapping issue

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Ballerina Version: 2201.4.1
* Operating System: Ubuntu 20.04
* Java SDK: 11

# Checklist:

### Changes to OpenAPI definition

- [ ] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [ ] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [ ] Added license header at the top of each .bal file. 
- [ ] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [ ] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [ ] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [ ] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [ ] Compiled the connector successfully with the targeted ballerina version. 
- [ ] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by NipunaRanasinghe — APPROVED** [review]

File: openapi/openai.audio/Ballerina.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ name = "openai.audio"
 icon = "icon.png"
 distribution = "2201.4.1"
 repository = "https://github.com/ballerina-platform/openapi-connectors/tree/main/openapi/openai.audio"
-version = "1.0.4"
+version = "1.0.5"
```
Comment (NipunaRanasinghe) [inline openapi/openai.audio/Ballerina.toml:9]: any reason to bump openAI connector versions?

File: openapi/shopify.admin/types.bal [hunk 2]
```diff
@@ -134,7 +134,7 @@ public type OrderAdjustment record {
     # The unique identifier for the refund that the order adjustment is associated with.
     int? refund_id?;
     # The value of the discrepancy between the calculated refund and the actual refund. If the kind property's value is shipping_refund, then amount returns the value of shipping charges refunded to the customer.
-    int? amount?;
+    string? amount?;
```
Comment (NipunaRanasinghe) [inline openapi/shopify.admin/types.bal:137]: Have we verified whether we should support both `int` and `string` (union), to be on the safe side? I'm seeing some amounts as integers in https://shopify.dev/docs/api/admin-rest/2023-07/resources/refund#resource-object

File: openapi/shopify.admin/openapi.yml [hunk 3]
```diff
@@ -3338,7 +3338,10 @@ components:
           type: integer
           description: The unique identifier for the refund that the order adjustment is associated with.
         amount:
-          type: integer
+          type: object
+          oneOf:
+            - $ref: '#/components/schemas/Integer'
+            - $ref: '#/components/schemas/String'
```
[the remark this answers is not in the document]
Comment (aneeshafedo) [reply openapi/shopify.admin/openapi.yml:3344]:
```suggestion
		oneOf:
            - type: integer
            - type: string
```

[the remark this answers is not in the document]
Comment (RolandHewage) [reply openapi/shopify.admin/openapi.yml]: This is fixed now. Thank you

**Review by NipunaRanasinghe — APPROVED** [review]

File: openapi/shopify.admin/openapi.yml [hunk 4]
```diff
@@ -3338,7 +3338,9 @@ components:
           type: integer
           description: The unique identifier for the refund that the order adjustment is associated with.
         amount:
-          type: integer
+          oneOf:
+            - type: integer
+            - type: string
```
Comment (aneeshafedo) [inline openapi/shopify.admin/openapi.yml:3343]: Just notice that the type you want to generate is nullable too. in such case you need to make the field nullable too. 

```suggestion
          oneOf:
            - type: integer
          	  nullable: true
            - type: string
          	  nullable: true
```

**Pull request merged**